tests/recorder/recorder.go

```diff
@@ -0,0 +1,233 @@
+package main
```

Comment:
This package is really cool! I have been suffering with the problem of how to test clis for close to half a year and this is really nice approach. How would you feel about making this an external package? I would love to use it for some projects i'm working on